### PR TITLE
[workloadmeta/containerd] Improve error handling when generating initial images events

### PR DESCRIPTION
**Pull request by davidor**
### What does this PR do?

Fixes an issue in the containerd workloadmeta collector when the container image collection is enabled.

The collector does not start when there is an error extracting the information for one of the images. With this PR, instead of returning an error, we log a warning and return the rest of images. This is the same strategy followed when generating the information for the initial containers.

### Describe how to test/QA your changes

Check our internal logs. The `workloadmeta collector "containerd" could not start` errors shouldn't be caused by errors while fetching the images.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/workloadmeta/collectors/internal/containerd/containerd.go [hunk 1]
```diff
@@ -271,7 +271,8 @@ func (c *collector) notifyInitialImageEvents(ctx context.Context, namespace stri
 
 	for _, image := range existingImages {
 		if err := c.notifyEventForImage(ctx, namespace, image, nil); err != nil {
-			return err
+			log.Warnf(err.Error())
```
Comment (clamoriniere) [inline pkg/workloadmeta/collectors/internal/containerd/containerd.go:274]: does the error is self explanatory? should add a small logs prefix to provide more context to the error?

Comment (davidor) [reply pkg/workloadmeta/collectors/internal/containerd/containerd.go]: I wrapped the error to add some extra information :+1:
The wrapped error already contains everything that we need to debug the problem.

**Review by clamoriniere — APPROVED** [review]

File: releasenotes/notes/workloadmeta-containerd-err-handling-initial-events-b08fa6537e11eddf.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Fixes a bug that prevents the containerd workloadmeta collector from
+    starting some times when `container_image_collection.metadata.enabled` is
```
Comment (alai97) [inline releasenotes/notes/workloadmeta-containerd-err-handling-initial-events-b08fa6537e11eddf.yaml:12]:
```suggestion
    starting sometimes when `container_image_collection.metadata.enabled` is
```

**Review by alai97 — APPROVED** [review]
Looks good, one small suggestion!

**Pull request merged**